### ISSUE TITLE
GKE NodePool locations list in google_container_node_pool

### ISSUE THREAD

**Issue by astorath** [terraform-providers/terraform-provider-google#3992]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Hi! Google beta GKE API do support `locations` array in `NodePool` resource. This gives an option to:
- allocate nodes in different zones, yet in any preferred number (not just proportional to number of zones in cluster)
- more gradually control node allocations

### New or Affected Resource(s)

* google_container_node_pool

### Potential Terraform Configuration

```tf
resource "google_container_node_pool" "primary_preemptible_nodes" {
  name       = "sandbox-node-pool-1"
  location   = "us-central"
  cluster    = "sandbox"
  node_count = 1
  node_locations = ["us-central1"]

  node_config {
    preemptible  = true
    machine_type = "n1-standard-1"
  }
}
```

### References

* https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters.nodePools#NodePool

**Comment by rileykarson** [comment]
I've got a PR staged that adds this, there was a blocking bug in the upstream API that caused me to hold it. It should be fixed in prod now / soon, this _should_ land in our next release.

**Comment by redbaron** [comment]
@rileykarson , do you know whether  upstream bug was resolved?

**Comment by rileykarson** [comment]
Not yet- it should land in the next GKE release, and from there we can include the field in our next Terraform release.

**Comment by redbaron** [comment]
@rileykarson , great news.  This can also be an opportunity to resolve coupling between nodepool location and cluster location. If your PR would use location as cluster location and node_locations to specify nodepool's location, then it would also resolve #4298

**Comment by rileykarson** [comment]
Yep- it's an issue asking for the same feature. Closed #4298 as a dupe.
